### PR TITLE
feat: windows support

### PR DESCRIPTION
**Pull request by korki43**
What i did to support windows: 
 - catch the errors that a not existing dir causes and serve the 404 site instead
 - use windows command `ipconfig` to detect the local ip.

I tested it on my windows 10 machine.
This closes #26

### CODE REVIEW

**Comment by korki43** [comment]
Will this be reviewed or merged in the near future? I heard no feedback.

**Comment by joakimunge** [comment]
> Will this be reviewed or merged in the near future? I heard no feedback.

Hi, currently I'm a bit busy with my full time job and other engagements, but will get to an actual review as soon as I can.

Thanks for submitting!

**Comment by korki43** [comment]
ok, thanks 😄

**Review by joakimunge — CHANGES_REQUESTED** [review]
Hi. Sorry for the late review. I have been very busy this last month. 

Great job on this. I added some minor comments. I would also ask you to introduce a Github Action workflow that runs the tests in a Windows Container to make sure the tests pass for windows as well.

Otherwise I can do that when I get the chance :)

**Comment by korki43** [comment]
I added a commit to remove the redundant code and opened #29 which introduces the CI workflow. I will also open a PR in [joakimunge/deno-local-ip](https://github.com/joakimunge/deno-local-ip) to synchronize the changes I made here.

File: mod.ts [hunk 1]
```diff
@@ -266,7 +271,8 @@ const setGlobals = async (args: DenoliverOptions): Promise<void> => {
       before = args.after
     } else {
       try {
-        const path = posix.resolve(`${root}/${args.after}`)
+        var path = resolve(`${root}/${args.after}`)
+        if (Deno.build.os == 'windows') path = 'file://'.concat(path)
```
Comment (joakimunge) [inline mod.ts:275]: I try to keep `var` and `let` out of the code whenever possible, and avoid mutation. I would suggest this approach instead.

```suggestion
        const path = Deno.build.os === "windows" ? 'file://'.concat(path) : resolve(`${root}/${args.after}`)
```

File: utils/local-ip.ts [hunk 2]
```diff
@@ -3,36 +3,52 @@
...
+import { decode, error } from './utils.ts'
 
 export const getNetworkAddr = async () => {
-  let ifconfig: Deno.Process | undefined
+  const isWin = Deno.build.os === 'windows'
+  const command = isWin ? 'ipconfig' : 'ifconfig'
   try {
-    ifconfig = await Deno.run({
-      cmd: ['ifconfig'],
+    let ifconfig = await Deno.run({
```
Comment (joakimunge) [inline utils/local-ip.ts:12]: Could do `const` here!

**Review by joakimunge — CHANGES_REQUESTED** [review]
First of all thank you again for working on this. I am sorry for the late comments. Very busy, hope you understand! 

I have some minor things I would like to adress before merging

**Comment by korki43** [comment]
I fully understand your points regarding the code style, but I'm currently unable to fix them as I unfortunately deleted my fork of your repo a while ago. I can't restore it via the "deleted Repos" tab, and I don't know another way to restore this branch to edit the code.

I think the repository owner is able to edit PRs. If this doesn't work, you could merge this PR and I (or you) would open a new PR to fix the code style issues. I also could close this PR and open a new one which will adress all the problems at once.

I'm really sorry for the inconvenience, I probably thougt this PR was closed when I deleted my fork.

**Comment by joakimunge** [comment]
> I fully understand your points regarding the code style, but I'm currently unable to fix them as I unfortunately deleted my fork of your repo a while ago. I can't restore it via the "deleted Repos" tab, and I don't know another way to restore this branch to edit the code.
> 
> I think the repository owner is able to edit PRs. If this doesn't work, you could merge this PR and I (or you) would open a new PR to fix the code style issues. I also could close this PR and open a new one which will adress all the problems at once.
> 
> I'm really sorry for the inconvenience, I probably thougt this PR was closed when I deleted my fork.

That's a shame! I will see what I can do to implement your changes without having you or me redo all the changes you have made. Thanks for the work on this!

**Comment by korki43** [comment]
superseded by #35

**Pull request closed without merging**